src/modules/module_27100.c

```diff
@@ -0,0 +1,407 @@
+
+  for (int i = 0; i < netntlm->user_len; i += 2)
+  {
+    out_buf[out_len++] = ptr[i];
+  }
+
+  out_buf[out_len++] = ':';
+  out_buf[out_len++] = ':';
+
+  ptr += netntlm->user_len;
+
+  for (int i = 0; i < netntlm->domain_len; i += 2)
+  {
+    out_buf[out_len++] = ptr[i];
+  }
+
+  out_buf[out_len++] = ':';
+
+  ptr = (u8 *) netntlm->chall_buf;
+
+  for (int i = 0; i < netntlm->srvchall_len; i++)
+  {
+    u8_to_hex (ptr[i], out_buf + out_len); out_len += 2;
+  }
+
+  out_buf[out_len++] = ':';
+
+  u32_to_hex (tmp[0], out_buf + out_len); out_len += 8;
+  u32_to_hex (tmp[1], out_buf + out_len); out_len += 8;
+  u32_to_hex (tmp[2], out_buf + out_len); out_len += 8;
+  u32_to_hex (tmp[3], out_buf + out_len); out_len += 8;
+
+  out_buf[out_len++] = ':';
+
+  ptr += netntlm->srvchall_len;
+
+  for (int i = 0; i < netntlm->clichall_len; i++)
+  {
+    u8_to_hex (ptr[i], out_buf + out_len); out_len += 2;
+  }
+
+  return out_len;
+}
+
+u32 module_pw_max (MAYBE_UNUSED const hashconfig_t *hashconfig, MAYBE_UNUSED const user_options_t *user_options, MAYBE_UNUSED const user_options_extra_t *user_options_extra)
+{
+  const u32 pw_max = 32; // Length of a NT hash
+
+  return pw_max;
+}
+
+u32 module_pw_min (MAYBE_UNUSED const hashconfig_t *hashconfig, MAYBE_UNUSED const user_options_t *user_options, MAYBE_UNUSED const user_options_extra_t *user_options_extra)
+{
+  const u32 pw_min = 32; // Length of a NT hash
+
+  return pw_min;
+}
+
+const char *module_benchmark_mask (MAYBE_UNUSED const hashconfig_t *hashconfig, MAYBE_UNUSED const user_options_t *user_options, MAYBE_UNUSED const user_options_extra_t *user_options_extra)
+{
+ const char *mask = "?a?a?a?a?a?a?a?axxxxxxxxxxxxxxxx";
+ return mask;
+}
+
+void module_init (module_ctx_t *module_ctx)
+{
+  module_ctx->module_context_size             = MODULE_CONTEXT_SIZE_CURRENT;
+  module_ctx->module_interface_version        = MODULE_INTERFACE_VERSION_CURRENT;
+
+  module_ctx->module_attack_exec              = module_attack_exec;
+  module_ctx->module_benchmark_esalt          = MODULE_DEFAULT;
+  module_ctx->module_benchmark_hook_salt      = MODULE_DEFAULT;
+  module_ctx->module_benchmark_mask           = module_benchmark_mask;
+  module_ctx->module_benchmark_salt           = MODULE_DEFAULT;
+  module_ctx->module_build_plain_postprocess  = MODULE_DEFAULT;
+  module_ctx->module_deep_comp_kernel         = MODULE_DEFAULT;
+  module_ctx->module_dgst_pos0                = module_dgst_pos0;
+  module_ctx->module_dgst_pos1                = module_dgst_pos1;
+  module_ctx->module_dgst_pos2                = module_dgst_pos2;
+  module_ctx->module_dgst_pos3                = module_dgst_pos3;
+  module_ctx->module_dgst_size                = module_dgst_size;
+  module_ctx->module_dictstat_disable         = MODULE_DEFAULT;
+  module_ctx->module_esalt_size               = module_esalt_size;
+  module_ctx->module_extra_buffer_size        = MODULE_DEFAULT;
+  module_ctx->module_extra_tmp_size           = MODULE_DEFAULT;
```

Comment:
Please add here:

```
  module_ctx->module_extra_tuningdb_block     = MODULE_DEFAULT;
``` 

This was added yesterday. Please also apply on the other module.

Reply:
done